src/lib.rs

```diff
@@ -623,18 +624,22 @@ fn generate_impl(
 	derive_where: &DeriveWhere,
 	trait_: &DeriveTrait,
 	item: &Item,
-	generics: &Generics,
+	generics: &SplitGenerics,
 ) -> TokenStream {
-	let (impl_generics, type_generics, where_clause) = generics.split_for_impl();
-	let mut where_clause = where_clause.map(Cow::Borrowed);
+	let SplitGenerics {
+		r#impl,
+		ty,
+		r#where,
```

Comment:
I'd probably have used something like `imp` and `wher` or `impl_gens` and `where_clause` instead of the `r#`.

Reply:
Done.